lib/improver/tests/nbhood/circular_kernel/test_GeneratePercentilesFromACircularNeighbourhood.py

```diff
@@ -0,0 +1,643 @@
+                [1., 1., 1., 1., 1.]]]]])
+        cube = self.cube
+        mask = np.zeros_like(cube.data)
+        mask[0, 0, 2, 2] = 1
+        cube.data = np.ma.masked_array(cube.data, mask=mask)
+        percentiles = np.array([10, 50, 90])
+        radius = 2000.
+        result = (
+            GeneratePercentilesFromACircularNeighbourhood(
+                percentiles=percentiles).run(
+                cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_masked_other_point(self):
+        """Test behaviour with a non-zero point next to a masked point.
+        The behaviour here is not right, as the mask is ignored."""
+        expected = np.array(
+            [[[[[1., 1., 1., 1., 1.],
+                [1., 1., 0.4, 1., 1.],
+                [1., 0.4, 0.4, 0.4, 1.],
+                [1., 1., 0.4, 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1.]]]]])
+        cube = self.cube
+        mask = np.zeros_like(cube.data)
+        mask[0, 0, 2, 3] = 1
+        cube.data = np.ma.masked_array(cube.data, mask=mask)
+        percentiles = np.array([10, 50, 90])
+        radius = 2000.
+        result = (
+            GeneratePercentilesFromACircularNeighbourhood(
+                percentiles=percentiles).run(
+                    cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_low_percentiles(self):
+        """Test behaviour with low percentiles."""
+        expected = np.array(
+            [[[[[1., 1., 1., 1., 1.],
+                [1., 1., 0.2, 1., 1.],
+                [1., 0.2, 0.2, 0.2, 1.],
+                [1., 1., 0.2, 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 0.4, 1., 1.],
+                [1., 0.4, 0.4, 0.4, 1.],
+                [1., 1., 0.4, 1., 1.],
+                [1., 1., 1., 1., 1.]]],
+              [[[1., 1., 1., 1., 1.],
+                [1., 1., 0.8, 1., 1.],
+                [1., 0.8, 0.8, 0.8, 1.],
+                [1., 1., 0.8, 1., 1.],
+                [1., 1., 1., 1., 1.]]]]])
+        cube = set_up_cube(
+            zero_point_indices=((0, 0, 2, 2),), num_time_points=1,
+            num_grid_points=5)
+        percentiles = np.array([5, 10, 20])
+        radius = 2000.
+        result = (
+            GeneratePercentilesFromACircularNeighbourhood(
+                percentiles=percentiles).run(
+                    cube, radius))
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_single_point_range_0(self):
```

Comment:
Does this test need to be here, seeing as its testing a helper function. I'm ok with it staying here though.

Reply:
I've left this here.